File: src/ValueDomain.cs
namespace Vim.Math3d
{
    public class ValueDomain
    {
        public readonly double Lower;
        public readonly double Upper;

        public ValueDomain(double lower, double upper)
            => (Lower, Upper) = (lower, upper);

        public double Normalize(double value)
            => value.Clamp(Lower, Upper) / Upper;
    }
}

File: dart/lib/src/value_domain.dart
part of '../vim_math3d.dart';

class ValueDomain {
  final double lower;
  final double upper;

  const ValueDomain([this.lower = 0.0, this.upper = 0.0]);

  double normalize(double value) {
    return value.clamp(lower, upper) / upper;
  }
}
